aci/resource_aci_cloudtemplateregiondetail.go

```diff
@@ -0,0 +1,167 @@
+			"parent_dn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"hub_networking": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"disabled",
+					"enabled",
+				}, false),
+			},
+		})),
+	}
+}
+
+func getRemoteCloudTemplateRegion(client *client.Client, dn string) (*models.CloudTemplateRegion, error) {
+	cloudTemplateRegionDetailCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	cloudTemplateRegionDetail := models.CloudTemplateRegionFromContainer(cloudTemplateRegionDetailCont)
+	if cloudTemplateRegionDetail.DistinguishedName == "" {
+		return nil, fmt.Errorf("Cloud Template Region %s not found", dn)
+	}
+	return cloudTemplateRegionDetail, nil
+}
+
+func setCloudTemplateRegionAttributes(cloudTemplateRegionDetail *models.CloudTemplateRegion, d *schema.ResourceData) (*schema.ResourceData, error) {
+	d.SetId(cloudTemplateRegionDetail.DistinguishedName)
+	cloudTemplateRegionDetailMap, err := cloudTemplateRegionDetail.ToMap()
+	if err != nil {
+		return d, err
+	}
+	if d.Id() != cloudTemplateRegionDetail.DistinguishedName {
+		d.Set("parent_dn", "")
+	} else {
+		d.Set("parent_dn", GetParentDn(cloudTemplateRegionDetail.DistinguishedName, fmt.Sprintf("/"+models.RnCloudtemplateRegionDetail)))
+	}
+	d.Set("annotation", cloudTemplateRegionDetailMap["annotation"])
+	d.Set("hub_networking", toggleOptions(cloudTemplateRegionDetailMap["hubNetworkingEnabled"]))
+	return d, nil
+}
+
+func resourceAciCloudTemplateRegionImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	dn := d.Id()
+	log.Printf("[DEBUG] %s: Beginning Import", dn)
+	aciClient := m.(*client.Client)
+	cloudTemplateRegionDetail, err := getRemoteCloudTemplateRegion(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setCloudTemplateRegionAttributes(cloudTemplateRegionDetail, d)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] %s: Import finished successfully", dn)
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func makeCloudTemplateRegion(ctx context.Context, d *schema.ResourceData, m interface{}, method string) diag.Diagnostics {
+	aciClient := m.(*client.Client)
+	cloudProviderandRegionNamesDn := d.Get("parent_dn").(string)
+
+	cloudTemplateRegionDetailAttr := models.CloudTemplateRegionAttributes{}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		cloudTemplateRegionDetailAttr.Annotation = Annotation.(string)
+	} else {
+		cloudTemplateRegionDetailAttr.Annotation = "{}"
+	}
+
+	if HubNetworkingEnabled, ok := d.GetOk("hub_networking"); ok {
+		cloudTemplateRegionDetailAttr.HubNetworkingEnabled = toggleOptions(HubNetworkingEnabled.(string))
+	}
+	cloudTemplateRegionDetail := models.NewCloudTemplateRegion(fmt.Sprintf(models.RnCloudtemplateRegionDetail), cloudProviderandRegionNamesDn, cloudTemplateRegionDetailAttr)
```

Comment:
fmt.Sprintf seems redundant

```suggestion
	cloudTemplateRegionDetail := models.NewCloudTemplateRegion(models.RnCloudtemplateRegionDetail, cloudProviderandRegionNamesDn, cloudTemplateRegionDetailAttr)
```

Reply:
ok